carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/P2Utils.java

```diff
@@ -0,0 +1,188 @@
+        int min = -1;
+        if (p2InfFile.exists()) {
+
+            try (InputStreamReader reader = new InputStreamReader(new FileInputStream(p2InfFile), DEFAULT_ENCODING);
+                 BufferedReader in = new BufferedReader(reader)) {
+                String line;
+                while ((line = in.readLine()) != null) {
+                    String[] split = line.split("=");
+                    String[] split2 = split[0].split(Pattern.quote("."));
+                    if (split2[0].equalsIgnoreCase("properties")) {
+                        int index = Integer.parseInt(split2[1]);
+                        if (index > min) {
+                            min = index;
+                        }
+                    }
+                }
+            }
+        }
+        return min;
+    }
+
+    public static boolean isMatchString(String matchStr) {
+        for (String match : matchList) {
+            if (matchStr.equalsIgnoreCase(match)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static String getMatchRule(String matchStr) {
+        if (isPatch(matchStr)) {
+            return "perfect";
+        }
+        for (String match : matchList) {
+            if (matchStr.equalsIgnoreCase(match)) {
+                return match;
+            }
+        }
+        return null;
+    }
+
+    public static boolean isPatch(String matchStr) {
+        return matchStr.equalsIgnoreCase("patch");
+    }
+
+    public static void createCategoryFile(MavenProject project, List categories, File categoryFile)
+            throws ParserConfigurationException, TransformerException, MojoExecutionException {
+
+        Map featureCategories = new HashMap();
+
+        Document doc = getManifestDocument();
+        Element rootElement = doc.getDocumentElement();
+
+        if (rootElement == null) {
+            rootElement = doc.createElement("site");
+            doc.appendChild(rootElement);
+        }
+
+        for (Object object : categories) {
+            if (object instanceof Category) {
+                Category cat = (Category) object;
+                Element categoryDef = doc.createElement("category-def");
+                categoryDef.setAttribute("name", cat.getId());
+                categoryDef.setAttribute("label", cat.getLabel());
+                rootElement.appendChild(categoryDef);
+                Element descriptionElement = doc.createElement("description");
+                descriptionElement.setTextContent(cat.getDescription());
+                categoryDef.appendChild(descriptionElement);
+                ArrayList<CatFeature> processedFeatures = cat.getProcessedFeatures(project);
+                for (CatFeature feature : processedFeatures) {
+                    if (!featureCategories.containsKey(feature.getId() + feature.getVersion())) {
+                        ArrayList list = new ArrayList();
```

Comment:
What is the type of this List? There are IDE generated warnings for unchecked assignments to this List.